File: app/controllers/Application.java
package controllers;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.node.ObjectNode;
import com.mongodb.DBCollection;

import models.Note;
import net.vz.mongodb.jackson.JacksonDBCollection;
import play.*;
import play.libs.Json;
import play.modules.mongodb.jackson.MongoDB;
import play.modules.mongodb.jackson.MongoDBPlugin;
import play.mvc.*;
import views.html.*;


public class Application extends Controller {

    public static Result index() {
    	JacksonDBCollection<Note, String> coll = MongoDB.collection(Note.class, String.class, play.api.Play.current());
    	
    	return ok(index.render("Your notes application is ready.",coll.find().toArray(),NoteController.dynamicForm));
    }
    

    public static Result jsonnotes() {
    	return ok(views.html.json.render());
    }

}

File: app/controllers/Application.scala
package controllers

import play.api.data.Form
import play.api.mvc._
import play.modules.reactivemongo._
import reactivemongo.bson._
import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.Future
import play.api.Play.current
import reactivemongo.play.json.collection.JSONCollection
import javax.inject.Inject
import play.api.libs.json._
import reactivemongo.play.json._
import reactivemongo.api.Cursor
import play.modules.reactivemongo._
import play.modules.reactivemongo.json.collection._
import reactivemongo.api.collections.bson.BSONCollection

class Application @Inject() (val reactiveMongoApi: ReactiveMongoApi)
  extends Controller with MongoController with ReactiveMongoComponents {

    lazy val coll =  db.collection[BSONCollection]("notes")

    def index() = Action.async { implicit response =>
      coll.find(BSONDocument()).cursor[models.Note].collect[List]().map{
          notes =>
            Ok(views.html.index("Your notes application is ready.")(notes))
      }
    }

    def jsonnotes() = Action {
    	Ok(views.html.json())
    }

}
